site/public/js/docker_interface.js

```diff
@@ -17,3 +17,122 @@ function collapseSection(id,btn_id) {
+                hide = false;
+            }
+        });
+        if (hide) {
+            this_row.hide();
+        }
+        else {
+            this_row.show();
+        }
+    });
+}
+
+function showAll() {
+    $('.image-row').show();
+    $('.filter-buttons').removeClass('fully-transparent');
+}
+
+function addFieldOnChange() {
+    const command = $(this).val();
+    const regex = new RegExp('^[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+/[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+:[a-zA-Z0-9][a-zA-Z0-9._-]{0,127}$');
+    if (!regex.test(command)) {
+        $('#send-button').attr('disabled',true);
+        if (command !== '') {
+            $('#docker-warning').css('display', '');
+            localStorage.setItem('capability', command);
+        }
+        else {
+            localStorage.removeItem('capability');
+        }
+    }
+    else {
+        $('#send-button').attr('disabled',false);
+        $('#docker-warning').css('display', 'none');
+        localStorage.setItem('capability', command);
+    }
+}
+
+function addImage(url) {
+    const capability = $('#capability-form').val();
+    const image = $('#add-field').val();
+    $.ajax({
+        url: url,
+        type: 'POST',
+        data: {
+            'capability': capability,
+            'image': image,
+            // eslint-disable-next-line no-undef
+            csrf_token: csrfToken,
+        },
+        success: function(data) {
+            const json = JSON.parse(data);
+            if (json.status == 'success') {
+                $('#add-field').val('');
+                localStorage.removeItem('capability');
+                // eslint-disable-next-line no-undef
+                displaySuccessMessage(json.data);
+            }
+            else {
+                // eslint-disable-next-line no-undef
+                displayErrorMessage(json.message);
+            }
+        },
+        error: function(err) {
+            console.error(err);
+            window.alert('Something went wrong. Please try again.');
+        },
+    });
+}
+
+function updateImage(url) {
+    $.ajax({
+        url: url,
+        type: 'GET',
+        data: {
+            // eslint-disable-next-line no-undef
+            csrf_token: csrfToken,
+        },
+        success: function(data) {
+            const json = JSON.parse(data);
+            if (json.status == 'success') {
+                // eslint-disable-next-line no-undef
+                displaySuccessMessage(json.data);
+            }
+            else {
+                // eslint-disable-next-line no-undef
+                displayErrorMessage(json.message);
+            }
+        },
+        error: function(err) {
+            console.error(err);
+            window.alert('Something went wrong. Please try again.');
+        },
+    });
+}
+
+$(document).ready(() => {
+    $('.filter-buttons').on('click', filterOnClick);
+    $('#show-all').on('click', showAll);
+    $('#add-field').val(localStorage.getItem('capability'));
```

Comment:
Why do we need the `localStorage` here? It looks like the only way to leave the page is the instructor themselves navigating away, at which point it feels like this `localStorage` would become stale.